Panic on wasm with 2.6.0-beta1

### Checklist

- [x] I have searched the [issue tracker](https://github.com/fyne-io/fyne/issues) for open issues that relate to the same problem, before opening a new one.
- [x] This issue only relates to a single bug. I will open new issues for any other problems.

### Describe the bug

Widgets that take a while to set up cause this race condition on wasm.

```
panic: syscall/js: call of Value.Call on null
wasm_exec.js:22 
wasm_exec.js:22 goroutine 1 [running]:
wasm_exec.js:22 syscall/js.Value.Call({{}, 0x7ff8000000000002, 0x0}, {0x19ca0b, 0x8}, {0x1433b38, 0x4, 0x4})
wasm_exec.js:22 	/usr/lib/go/src/syscall/js/js.go:438 +0x3f
wasm_exec.js:22 github.com/fyne-io/gl-js.Viewport(...)
wasm_exec.js:22 	/home/hayden/Development/Go/pkg/mod/github.com/fyne-io/gl-js@v0.1.0/gl_webgl_wasm.go:824
wasm_exec.js:22 fyne.io/fyne/v2/internal/painter/gl.(*xjsContext).Viewport(0x12b1140, 0x0, 0x0, 0xf00, 0x4d6)
wasm_exec.js:22 	/home/hayden/Development/Go/src/github.com/hkparker/fyne/internal/painter/gl/gl_wasm.go:238 +0x4
wasm_exec.js:22 fyne.io/fyne/v2/internal/painter/gl.(*painter).SetOutputSize(0x15aa700, 0xf00, 0x4d6)
wasm_exec.js:22 	/home/hayden/Development/Go/src/github.com/hkparker/fyne/internal/painter/gl/painter.go:103 +0x6
wasm_exec.js:22 fyne.io/fyne/v2/internal/driver/glfw.updateGLContext(0x1400fc0)
wasm_exec.js:22 	/home/hayden/Development/Go/src/github.com/hkparker/fyne/internal/driver/glfw/loop.go:228 +0x1d
wasm_exec.js:22 fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).repaintWindow.func1()
wasm_exec.js:22 	/home/hayden/Development/Go/src/github.com/hkparker/fyne/internal/driver/glfw/loop.go:197 +0xa
wasm_exec.js:22 fyne.io/fyne/v2/internal/driver/glfw.(*window).RunWithContext(0x1400fc0, 0x1433cc0)
wasm_exec.js:22 	/home/hayden/Development/Go/src/github.com/hkparker/fyne/internal/driver/glfw/window.go:899 +0x18
wasm_exec.js:22 fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).repaintWindow(0x15e3d90, 0x1400fc0)
wasm_exec.js:22 	/home/hayden/Development/Go/src/github.com/hkparker/fyne/internal/driver/glfw/loop.go:191 +0x4
wasm_exec.js:22 fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).drawSingleFrame(0x15e3d90)
wasm_exec.js:22 	/home/hayden/Development/Go/src/github.com/hkparker/fyne/internal/driver/glfw/loop.go:89 +0x1b
wasm_exec.js:22 fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).runGL(0x15e3d90)
wasm_exec.js:22 	/home/hayden/Development/Go/src/github.com/hkparker/fyne/internal/driver/glfw/loop.go:167 +0x76
wasm_exec.js:22 fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).Run(0x15e3d90)
wasm_exec.js:22 	/home/hayden/Development/Go/src/github.com/hkparker/fyne/internal/driver/glfw/driver.go:162 +0x6
wasm_exec.js:22 fyne.io/fyne/v2/app.(*fyneApp).Run(0x15e3e40)
wasm_exec.js:22 	/home/hayden/Development/Go/src/github.com/hkparker/fyne/app/app.go:72 +0xf
wasm_exec.js:22 fyne.io/fyne/v2/internal/driver/glfw.(*window).ShowAndRun(0x1400fc0)
wasm_exec.js:22 	/home/hayden/Development/Go/src/github.com/hkparker/fyne/internal/driver/glfw/window.go:221 +0xc
wasm_exec.js:22 main.main()
wasm_exec.js:22 	/home/hayden/Development/Go/src/github.com/hkparker/fyne-dev/main.go:82 +0x3a
```

### How to reproduce

`fyne serve` the example code and click the button while looking at the javascript console

### Screenshots

_No response_

### Example code

```go
package main

import (
	"fyne.io/fyne/v2"
	"fyne.io/fyne/v2/app"
	"fyne.io/fyne/v2/container"
	"fyne.io/fyne/v2/widget"
	"time"
)

type slowLabel struct {
	widget.Label
}

func newSlowLabel(str string) *slowLabel {
	sl := &slowLabel{
		Label: widget.Label{Text: str},
	}
	sl.ExtendBaseWidget(sl)

	return sl
}

func (sl *slowLabel) MinSize() fyne.Size {
	time.Sleep(50 * time.Millisecond)
	return sl.Label.MinSize()
}

func main() {
	a := app.New()
	w := a.NewWindow("crash")

	var mainContent fyne.CanvasObject
	mainContent = container.NewVBox(
		newSlowLabel("click this button to crash"),
		widget.NewButton("Crash", func() {
			w.SetContent(mainContent)
		}))

	w.SetContent(mainContent)

	w.ShowAndRun()
}
```

### Fyne version

develop

### Go compiler version

go 1.24.1 linux/amd64

### Operating system and version

Arch Linux / Chrome

### Additional Information

_No response_

--- a/internal/async/goroutine_desktop.go
+++ b/internal/async/goroutine_desktop.go
@@ -1,4 +1,4 @@
-//go:build !wasm && !mobile
+//go:build !mobile
 
 package async
 

--- a/internal/async/goroutine_wasm.go
+++ b/internal/async/goroutine_wasm.go
@@ -1,8 +1,0 @@
-//go:build wasm
-
-package async
-
-// IsMainGoroutine returns true if it is called from the main goroutine, false otherwise.
-func IsMainGoroutine() bool {
-	return true
-}

--- a/internal/driver/glfw/loop.go
+++ b/internal/driver/glfw/loop.go
@@ -86,7 +86,9 @@ func (d *gLDriver) drawSingleFrame() {
 			continue
 		}
 
-		refreshed = refreshed || d.repaintWindow(w)
+		w.RunWithContext(func() {
+			refreshed = refreshed || w.driver.repaintWindow(w)
+		})
 	}
 	cache.Clean(refreshed)
 }
@@ -188,26 +190,25 @@ func (d *gLDriver) destroyWindow(w *window, index int) {
 func (d *gLDriver) repaintWindow(w *window) bool {
 	canvas := w.canvas
 	freed := false
-	w.RunWithContext(func() {
-		if canvas.EnsureMinSize() {
-			w.shouldExpand = true
-		}
-		freed = canvas.FreeDirtyTextures() > 0
+	if canvas.EnsureMinSize() {
+		w.shouldExpand = true
+	}
+	freed = canvas.FreeDirtyTextures() > 0
 
-		updateGLContext(w)
-		canvas.paint(canvas.Size())
+	updateGLContext(w)
+	canvas.paint(canvas.Size())
 
-		view := w.viewport
-		visible := w.visible
+	view := w.viewport
+	visible := w.visible
 
-		if view != nil && visible {
-			view.SwapBuffers()
-		}
+	if view != nil && visible {
+		view.SwapBuffers()
+	}
 
-		// mark that we have walked the window and don't
-		// need to walk it again to mark caches alive
-		w.lastWalkedTime = time.Now()
-	})
+	// mark that we have walked the window and don't
+	// need to walk it again to mark caches alive
+	w.lastWalkedTime = time.Now()
+	w.lastWalkedTime = time.Now()
 	return freed
 }
 

--- a/internal/driver/glfw/loop.go
+++ b/internal/driver/glfw/loop.go
@@ -208,7 +208,6 @@ func (d *gLDriver) repaintWindow(w *window) bool {
 	// mark that we have walked the window and don't
 	// need to walk it again to mark caches alive
 	w.lastWalkedTime = time.Now()
-	w.lastWalkedTime = time.Now()
 	return freed
 }
 

--- a/internal/driver/glfw/window.go
+++ b/internal/driver/glfw/window.go
@@ -305,7 +305,9 @@ func (w *window) processResized(width, height int) {
 		return
 	}
 
-	w.platformResize(canvasSize)
+	w.RunWithContext(func() {
+		w.platformResize(canvasSize)
+	})
 }
 
 func (w *window) processFrameSized(width, height int) {

--- a/internal/driver/glfw/window_wasm.go
+++ b/internal/driver/glfw/window_wasm.go
@@ -156,26 +156,36 @@ func (w *window) detectScale() float32 {
 }
 
 func (w *window) moved(_ *glfw.Window, x, y int) {
-	w.processMoved(x, y)
+	runOnMain(func() {
+		w.processMoved(x, y)
+	})
 }
 
 func (w *window) resized(_ *glfw.Window, width, height int) {
-	w.canvas.scale = w.calculatedScale()
-	w.processResized(width, height)
+	runOnMain(func() {
+		w.canvas.scale = w.calculatedScale()
+		w.processResized(width, height)
+	})
 }
 
 func (w *window) frameSized(_ *glfw.Window, width, height int) {
-	w.processFrameSized(width, height)
+	runOnMain(func() {
+		w.processFrameSized(width, height)
+	})
 }
 
 func (w *window) refresh(_ *glfw.Window) {
-	w.processRefresh()
+	runOnMain(func() {
+		w.processRefresh()
+	})
 }
 
 func (w *window) closed(viewport *glfw.Window) {
-	viewport.SetShouldClose(false) // reset the closed flag until we check the veto in processClosed
+	runOnMain(func() {
+		viewport.SetShouldClose(false) // reset the closed flag until we check the veto in processClosed
 
-	w.processClosed()
+		w.processClosed()
+	})
 }
 
 func fyneToNativeCursor(cursor desktop.Cursor) (*Cursor, bool) {
@@ -189,24 +199,30 @@ func (w *window) setCustomCursor(rawCursor *Cursor, isCustomCursor bool) {
 }
 
 func (w *window) mouseMoved(_ *glfw.Window, xpos, ypos float64) {
-	w.processMouseMoved(w.scaleInput(xpos), w.scaleInput(ypos))
+	runOnMain(func() {
+		w.processMouseMoved(w.scaleInput(xpos), w.scaleInput(ypos))
+	})
 }
 
 func (w *window) mouseClicked(viewport *glfw.Window, btn glfw.MouseButton, action glfw.Action, mods glfw.ModifierKey) {
-	button, modifiers := convertMouseButton(btn, mods)
-	mouseAction := convertAction(action)
+	runOnMain(func() {
+		button, modifiers := convertMouseButton(btn, mods)
+		mouseAction := convertAction(action)
 
-	w.processMouseClicked(button, mouseAction, modifiers)
+		w.processMouseClicked(button, mouseAction, modifiers)
+	})
 }
 
 func (w *window) mouseScrolled(viewport *glfw.Window, xoff, yoff float64) {
-	if runtime.GOOS != "darwin" && xoff == 0 &&
-		(viewport.GetKey(glfw.KeyLeftShift) == glfw.Press ||
-			viewport.GetKey(glfw.KeyRightShift) == glfw.Press) {
-		xoff, yoff = yoff, xoff
-	}
+	runOnMain(func() {
+		if runtime.GOOS != "darwin" && xoff == 0 &&
+			(viewport.GetKey(glfw.KeyLeftShift) == glfw.Press ||
+				viewport.GetKey(glfw.KeyRightShift) == glfw.Press) {
+			xoff, yoff = yoff, xoff
+		}
 
-	w.processMouseScrolled(xoff, yoff)
+		w.processMouseScrolled(xoff, yoff)
+	})
 }
 
 func convertMouseButton(btn glfw.MouseButton, mods glfw.ModifierKey) (desktop.MouseButton, fyne.KeyModifier) {

